aten/src/ATen/native/TensorCompare.cpp

```diff
@@ -113,14 +113,16 @@ Tensor isclose(const Tensor& self, const Tensor& other, double rtol, double atol
   // by the default scalar type then this may cause an incorrect result.
 
   // Computes allowed and actual error
-  Tensor cast_other;
+  Tensor cast_self, cast_other;
   if (c10::isIntegralType(self.scalar_type(), /*includeBool=*/true)) {
+    cast_self = self.to(at::get_default_dtype());
     cast_other = other.to(at::get_default_dtype());
```

Comment:
#60533 says that we want to cast the operands to the most precise datatype in the relevant category (integer here), is that what `at::get_default_dtype()` does?

Reply:
`at::get_default_dtype()` here would return a `float` dtype, although I know it's not the most accurate data type, but it's just consistent with the previous practice, including the upcasting of `uint8` tensors mentioned in #60533.
https://github.com/pytorch/pytorch/blob/94840969e48a228ed19a0289ebc55ce04f83f7ad/c10/core/DefaultDtype.cpp#L5-L6